orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestEvaluator.java

```diff
@@ -16,172 +16,245 @@
+  }
 
-      requiredArtifacts.add(requiredArtifact);
+  private ImmutableList<Map<Object, Object>> coerceManifestToList(Object manifest) {
+    if (manifest instanceof List) {
+      return ImmutableList.copyOf(
+          objectMapper.convertValue(manifest, new TypeReference<List<Map<Object, Object>>>() {}));
     }
+    Map<Object, Object> singleManifest =
+        (Map<Object, Object>) objectMapper.convertValue(manifest, Map.class);
+    return ImmutableList.of(singleManifest);
+  }
 
-    // resolve SpEL expressions in artifacts defined inline in the stage
-    for (ManifestContext.BindArtifact artifact :
-        Optional.ofNullable(context.getRequiredArtifacts()).orElse(emptyList())) {
-      Artifact requiredArtifact =
-          artifactUtils.getBoundArtifactForStage(
-              stage, artifact.getExpectedArtifactId(), artifact.getArtifact());
-
-      if (requiredArtifact == null) {
-        throw new IllegalStateException(
-            "No artifact with id '"
-                + artifact.getExpectedArtifactId()
-                + "' could be found in the pipeline context.");
+  private Supplier<Iterable<Object>> fetchAndParseManifestYaml(Artifact manifestArtifact) {
+    return () -> {
+      Response manifestText = oortService.fetchArtifact(manifestArtifact);
+      try (InputStream body = manifestText.getBody().in()) {
+        return yamlParser.get().loadAll(body);
+      } catch (Exception e) {
+        throw new IllegalStateException(e);
       }
+    };
+  }
+
+  private Artifact getManifestArtifact(Stage stage, ManifestContext context) {
+    Artifact manifestArtifact =
+        Optional.ofNullable(
+                artifactUtils.getBoundArtifactForStage(
+                    stage, context.getManifestArtifactId(), context.getManifestArtifact()))
+            .orElseThrow(() -> new IllegalArgumentException("No manifest artifact was specified."));
 
-      requiredArtifacts.add(requiredArtifact);
+    // Once the legacy artifacts feature is removed, all trigger expected artifacts will be
+    // required to define an account up front.
+    if (context.getManifestArtifactAccount() != null) {
+      manifestArtifact.setArtifactAccount(context.getManifestArtifactAccount());
     }
 
-    log.info("Artifacts {} are bound to the manifest", requiredArtifacts);
+    if (manifestArtifact.getArtifactAccount() == null) {
+      throw new IllegalArgumentException("No manifest artifact account was specified.");
+    }
+
+    return manifestArtifact;
+  }
+
+  private ImmutableList<Map<Object, Object>> getSpelEvaluatedManifests(
+      ImmutableList<Map<Object, Object>> unevaluatedManifests, Stage stage) {
+    Map<String, Object> manifestWrapper = new HashMap<>();
+    manifestWrapper.put("manifests", unevaluatedManifests);
+
+    manifestWrapper =
+        contextParameterProcessor.process(
+            manifestWrapper, contextParameterProcessor.buildExecutionContext(stage), true);
```

Comment:
What is `true`? Not in a philosophical sense, but more like what is that parameter doing? go/java-practices/comments#arguments suggests annotating it for readers with `/* allowUnknownKeys= */ true`. (If you put your cursor on `true` and hit `ctrl+alt+/` it should automatically add that comment.)

This is super subjective, but I also don't know about reusing `manifestWrapper`. It feels like you have two different variables that coincidentally have the same type, but are conceptually different things? I'd be inclined to call one `processorInput` and `processorResult`? I know nothing about the `ContextParameterProcessor` so I definitely may be wrong here.

Reply:
Still don't know what is philosophically `true`, but added the arguments annotation so folks can at least try to understand what `allowUnknownKeys: true` means (comes from the [expression logic in Kork](https://github.com/spinnaker/kork/blob/master/kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/whitelisting/MapPropertyAccessor.java#L28) which I don't really understand the purpose of to be honest).

Good call on distinguishing input vs. result, updated that as well!